### ISSUE TITLE
Restructure Flask Applications into blueprints

### ISSUE THREAD

**Issue by redshiftzero** [freedomofpress/securedrop#2294]
## Description

*Note:* This is only to be done once internationalization of the Flask code is complete.

Our source and journalist Flask applications (`source.py` and `journalist.py` and their associated test suites) have become disorganized. It would simplify maintenance of the Python code to adopt a more standard structure for our Flask application and to consider leveraging [Flask's Blueprints](http://flask.pocoo.org/docs/0.12/blueprints/) (Flask's equivalent of Django's apps) to do so. 

A structure as follows would be a good start: 

```
journalist.py
source.py
journo_app/
    tests/
    views.py
    utils.py
admin_app/
    tests/
    views.py
    utils.py
source_app/
    tests/
    views.py
    utils.py
db.py
```

In this configuration, we would register the admin and journalist application blueprints in `journalist.py` (for the admin blueprint, on the URL prefix `admin/`), and the source blueprint in `source.py` using Flask's `register_blueprint`. 

For each application, we should break up the code into _at least_ `views.py` and `utils.py`. As a general rule, we should be moving logic out of our view functions and putting it into our models (`db.py`) or utility functions (`utils.py`) in the relevant application (to be fair, generic `utils.py` files  can also get quite messy so at some point further organizing might be warranted). We should implement functionality like the journalist API (#1761) with minimal code duplication by using the utility functions (see for reference the API implemented _without_ blueprints [here](https://github.com/freedomofpress/securedrop/compare/journalist-api)). 

Related: #2057

## User Stories

As a SecureDrop developer, I want my codebase to be organized and structured to decrease cognitive load while making changes. 

As a developer familiar with other Flask applications but new to the SecureDrop codebase, I want to see familiar Flask patterns such that my time to spin up on the codebase is shorter.

**Comment by heartsucker** [comment]
Just as a note, I think not only is #2057 related, but I think it is a hard dependency of this.

**Comment by heartsucker** [comment]
More notes. I think we should strip down `journalist.py` and `source.py` into something minimal with just a main method because once we containerize our image, we're going to have to use some WSGI runner like `gunicorn` inside the containers (see #1389). The should be `{journalist,source}_app.py` and we should have modules `journalist` and `source` that hide most things aside from the `create_app` function via `__init__.py`.

**Comment by heartsucker** [comment]
This was completed for the source interface in #2365.

**Comment by heartsucker** [comment]
closed by #2466
